backendv2/pkg/license/detect.go

```diff
@@ -0,0 +1,301 @@
+	}
+
+	span.SetAttributes(attribute.Int("license.detected_count", len(allMatches)))
+
+	// Group matches by license for logging (similar to old behavior)
+	licenseGroups := make(map[string][]licensedb.Match)
+	for _, match := range allMatches {
+		licenseGroups[match.License] = append(licenseGroups[match.License], match)
+	}
+
+	slog.DebugContext(ctx, "License detection completed", "detected_licenses_count", len(allMatches))
+	return allMatches, nil
+}
+
+func (d detector) buildLicenseFileMap(matches []licensedb.Match, repoURL string) map[string][]License {
+	filesWithLicenses := make(map[string][]License)
+
+	for _, match := range matches {
+		// Skip deprecated licenses (this is license names, not filenames)
+		if strings.HasPrefix(match.License, "deprecated_") {
+			continue
+		}
+
+		_, isCompatible := d.licenseMap[strings.ToLower(match.License)]
+
+		if match.Confidence >= d.config.ConfidenceThreshold {
+			filesWithLicenses[match.File] = append(filesWithLicenses[match.File], License{
+				SPDX:         match.License,
+				Confidence:   match.Confidence,
+				IsCompatible: isCompatible,
+				File:         match.File,
+				Link:         generateGitHubLink(repoURL, match.File),
+			})
+		}
+	}
+
+	return filesWithLicenses
+}
+
+func (d detector) filterAndSortLicenseFiles(ctx context.Context, filesWithLicenses map[string][]License) []string {
+	var licenseFiles []string
+	for file := range filesWithLicenses {
+		if shouldIgnore, reason := shouldIgnoreLicenseFile(file); shouldIgnore {
+			slog.DebugContext(ctx, "Ignoring license file", "file", file, "reason", reason)
+			delete(filesWithLicenses, file)
+			continue
+		}
+		licenseFiles = append(licenseFiles, file)
+
+		// Sort licenses within each file by confidence
+		slices.SortFunc(filesWithLicenses[file], func(a, b License) int {
+			if a.Confidence > b.Confidence {
+				return -1
+			}
+			if a.Confidence < b.Confidence {
+				return 1
+			}
+			return strings.Compare(strings.ToLower(a.SPDX), strings.ToLower(b.SPDX))
+		})
+	}
+
+	// Sort license files: docs first, then path depth, then alphabetical
+	slices.SortFunc(licenseFiles, func(a, b string) int {
+		aIsDoc := isDocumentationDirectory(a)
+		bIsDoc := isDocumentationDirectory(b)
+
+		if aIsDoc != bIsDoc {
+			if aIsDoc {
+				return -1
+			}
+			return 1
+		}
+
+		aDepth := pathDepth(a)
+		bDepth := pathDepth(b)
+		if aDepth != bDepth {
+			return aDepth - bDepth
+		}
+
+		return strings.Compare(strings.ToLower(a), strings.ToLower(b))
+	})
+
+	return licenseFiles
+}
+
+func (d detector) collectResults(ctx context.Context, span trace.Span, licenseFiles []string, filesWithLicenses map[string][]License) []License {
+	var result []License
+
+	// Iterate through sorted list of potential license files
+	for _, file := range licenseFiles {
+		for _, l := range filesWithLicenses[file] {
+			// Exit early (keeping in mind the sort order above)
+			if l.Confidence >= d.config.ConfidenceOverrideThreshold {
```

Comment:
Do we want to store all the results in the DB and filter there instead?

Reply:
Hmm, maybe I could store all results over a certain threshold, thats not a bad idea.